lib/settings/pages/post_appearance_settings_page.dart

```diff
@@ -511,6 +540,35 @@ class _PostAppearanceSettingsPageState extends State<PostAppearanceSettingsPage>
               highlightKey: settingToHighlight == LocalSettings.dimReadPosts ? settingToHighlightKey : null,
             ),
           ),
+          SliverToBoxAdapter(
+            child: ToggleOption(
+              description: l10n.showFullDate,
+              subtitle: l10n.showFullDateDescription,
+              value: showFullPostDate,
+              iconEnabled: Icons.date_range_rounded,
+              iconDisabled: Icons.date_range_outlined,
+              onToggle: (bool value) => setPreferences(LocalSettings.showFullPostDate, value),
+              highlightKey: settingToHighlight == LocalSettings.showFullPostDate ? settingToHighlightKey : null,
+            ),
+          ),
+          SliverToBoxAdapter(
+            child: ListOption(
+              description: l10n.dateFormat,
+              disabled: !showFullPostDate,
+              value: ListPickerItem(
```

Comment:
For some reason the initially selected value is not shown as highlighted in the list the first time you open the modal. Once you pick a value, subsequently it is highlighted. So I think you'd have to clear your preferences to repro. But it's visible in your demo.

Maybe it's because the `DEFAULT_DATE_FORMAT` constant is not a reference match for any of the entries in the `dateFormats` list.